### PR TITLE
added functionality to handle RGBA, i.e. 4 channels and related modif…

### PR DESCRIPTION
**Pull request by bhavuk2002**
#12 

thanks. im new to open source, first i read about image channels and got the understanding what they are, here basically we have to add ALPHA in loading image, we are currently missing that, thats why errors are getting generaated. I read documentation for stbi_load and have modified the code base accordingly and handled error accordingly. 

Can someone test it?

### CODE REVIEW

**Review by YavorGIvanov — CHANGES_REQUESTED** [review]
There a couple of places in the example where the number of channels is assumed to be 3. This surely won't work. You will have to propagate the number of channels as a member variable in the sam_image_u8 and use it later.

You may need some help with this.

**Comment by bhavuk2002** [comment]
oh ya i was thinking of editing sam_image_u8 but neglected it. will do surely, and update sam_image_u8.

**Comment by bhavuk2002** [comment]
one more thing apart from that logic for load_image_from_file is fine?

**Comment by bhavuk2002** [comment]
image
in the code base here at line number 340 we are resizing the mask. Do we need to use 4 channels here or 3 are enough for 4 channels too? Apart from this function in main.cpp / examples, there is no hard coding of number of channels left.

**Comment by bhavuk2002** [comment]
can you please elaborate, how can i work forward on this?

**Comment by YavorGIvanov** [comment]
Should a png image with alpha channel and test that the sample works. I will try to do that today or tomorrow and look more carefully through your change.

**Comment by bhavuk2002** [comment]
uhm png image doesnt support alpha channel.

**Comment by YavorGIvanov** [comment]
https://www.w3.org/TR/PNG-DataRep.html Section 2.4

**Comment by bhavuk2002** [comment]
Okay i was wrong, now what changes should i make and where?

**Comment by bhavuk2002** [comment]
any info or update?

**Comment by bhavuk2002** [comment]
any update?

File: examples/main.cpp [hunk 1]
```diff
@@ -107,20 +107,30 @@ static bool downscale_img_to_screen(sam_image_u8 &img, SDL_Window* window) {
 
 static bool load_image_from_file(const std::string & fname, sam_image_u8 & img) {
     int nx, ny, nc;
-    auto data = stbi_load(fname.c_str(), &nx, &ny, &nc, 3);
-    if (!data) {
-        fprintf(stderr, "%s: failed to load '%s'\n", __func__, fname.c_str());
+
+    // instead of directly loading the using stbi_load, lets first check if image is in supported format or not
+    int ok = stbi_info(filename, &nx, &ny, &nc);
```
Comment (YavorGIvanov) [inline examples/main.cpp:112]: This doesn't compile. filename is undefined. You also added a couple of trailing tabs below.

File: examples/main.cpp [hunk 2]
```diff
@@ -107,20 +107,30 @@ static bool downscale_img_to_screen(sam_image_u8 &img, SDL_Window* window) {
...
+    if(ok == 0) return 0;
```
Comment (YavorGIvanov) [inline examples/main.cpp:113]: space after if

**Pull request closed without merging**